### PR TITLE
Add new hpa and pdb versions for samples

### PR DESCRIPTION
**Pull request by richardwxn**
Based on deprecation notice: https://kubernetes.io/docs/reference/using-api/deprecation-guide/.

Similar changes are added in OSS Istio.

I will add similar change to egress gateway part if this looks good

### CODE REVIEW

**Comment by richardwxn** [comment]
@zerobfd Do you think whether this layout looks ok?

**Review by zerobfd — APPROVED** [review]
LGTM, let's make sure this doesn't change any of the commands that we use in Cloud docs when referring to the samples.

File: samples/gateways/istio-ingressgateway/autoscaling-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Optional: HorizontalPodAutoscaler will automatically scale the gateway replica count based on
+# Used on >=1.23 K8S versions
+# CPU utilization
+apiVersion: autoscaling/v2beta1
```
Comment (howardjohn) [inline samples/gateways/istio-ingressgateway/autoscaling-v2.yaml:4]: shouldn't this be `v2`?

Comment (richardwxn) [reply samples/gateways/istio-ingressgateway/autoscaling-v2.yaml]: fixed

**Comment by richardwxn** [comment]
@zerobfd Can you help to merge it? Thanks

**Pull request merged**